src/overlays/actors/ovl_En_Fhg_Fire/z_en_fhg_fire.c

```diff
@@ -21,27 +45,492 @@ const ActorInit En_Fhg_Fire_InitVars = {
     (ActorFunc)EnFhgFire_Update,
     (ActorFunc)EnFhgFire_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fhg_Fire/func_80A0F260.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fhg_Fire/EnFhgFire_Init.s")
+void EnFhgFire_SetupAction(EnFhgFire* this, EnFhgFireActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnFhgFire_Init(Actor* thisx, GlobalContext* globalCtx) {
+    f32 tempf0;
+    EnFhgFire* this = THIS;
+    Player* player = PLAYER;
+    f32 tempf1;
+    f32 tempf2;
+    f32 tempf3;
+
+    ActorShape_Init(&thisx->shape, 0.0f, NULL, 0.0f);
+    if ((thisx->params == 0x23) || (thisx->params == 0x24) || (thisx->params == 0x32)) {
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, thisx, &D_80A11790);
+    }
+    this->unk_160 = 200.0f;
+    Actor_SetScale(thisx, 0.0f);
+
+    if (thisx->params == 0x01) {
+        EnFhgFire_SetupAction(this, func_80A0F6F8);
+        Audio_PlayActorSound2(thisx, NA_SE_EN_FANTOM_THUNDER);
+    } else if (thisx->params >= 0x64) {
+        EnFhgFire_SetupAction(this, func_80A0FA90);
+        thisx->shape.rot = thisx->posRot.rot;
+    }
+
+    if (thisx->params == 0x23) {
+        thisx->draw = NULL;
+        EnFhgFire_SetupAction(this, func_80A0FC48);
+        thisx->speedXZ = 30.0f;
+        Audio_PlayActorSound2(thisx, NA_SE_EN_FANTOM_SPARK);
+        return;
+    }
+
+    if (thisx->params == 0x24) {
+        EnFhgFire_SetupAction(this, func_80A0FD8C);
+        this->unk_160 = 255.0f;
+        this->unk_150.x = 0x20;
+        this->unk_150.y = 0x32;
+        this->unk_1FE = 0x0A;
+
+        tempf2 = thisx->posRot.rot.x;
+        this->unk_18C = tempf2 / 100.0f;
+        tempf1 = tempf2 * 0.13f;
+
+        this->collider.dim.radius = tempf1;
+        this->collider.dim.height = tempf1;
+        this->collider.dim.yShift = 0;
+        return;
+    }
+
+    if (thisx->params == 0x26) {
+        osSyncPrintf("yari hikari ct 1\n"); // "light spear"
```

Comment:
in general we put translation comments on top instead of to the side. just a small thing